src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  let pushObj;
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(newState, action.extraData);
+        pushObj = { ...newState };
+        result.push(pushObj);
+        break;
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete newState[key];
+        }
+        pushObj = { ...newState };
+        result.push(pushObj);
+        break;
+      case 'clear':
+        for (const key in newState) {
+          if (newState.hasOwnProperty(key)) {
+            delete newState[key];
+          }
+        }
+        pushObj = { ...newState };
+        result.push(pushObj);
+        break;
+      default:
+        throw new Error('Something went wrong !');
+    }
+  }
```

Comment:
you can also spread state clone into result array
```suggestion
  }
  result.push({ ...newState })
```